test/e2e/kubevirt.go

```diff
@@ -368,31 +369,33 @@ passwd:
 				return err
 			}).WithPolling(time.Second).WithTimeout(time.Minute).Should(Succeed())
 
-			Eventually(func() *kubevirtv1.VirtualMachineInstanceMigrationState {
-				err := crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)
-				Expect(err).ToNot(HaveOccurred())
-				return vmi.Status.MigrationState
-			}).WithOffset(1).WithPolling(time.Second).WithTimeout(10*time.Minute).ShouldNot(BeNil(), "should have a MigrationState")
-			Eventually(func() string {
-				err := crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)
-				Expect(err).ToNot(HaveOccurred())
-				return vmi.Status.MigrationState.TargetNode
-			}).WithOffset(1).WithPolling(time.Second).WithTimeout(10*time.Minute).ShouldNot(Equal(currentNode), "should refresh MigrationState")
-			Eventually(func() bool {
-				err := crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)
-				Expect(err).ToNot(HaveOccurred())
-				return vmi.Status.MigrationState.Completed
-			}).WithOffset(1).WithPolling(time.Second).WithTimeout(20*time.Minute).Should(BeTrue(), "should complete migration")
-			err = crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)
-			Expect(err).WithOffset(1).ToNot(HaveOccurred(), "should success retrieving vmi after migration")
-			Expect(vmi.Status.MigrationState.Failed).WithOffset(1).To(BeFalse(), func() string {
-				vmiJSON, err := json.Marshal(vmi)
-				if err != nil {
-					return fmt.Sprintf("failed marshaling migrated VM: %v", vmiJSON)
-				}
-				return fmt.Sprintf("should live migrate successfully: %s", string(vmiJSON))
-			})
-			Expect(vmi.Status.MigrationState.Mode).WithOffset(1).To(Equal(migrationMode), "should be the expected migration mode %s", migrationMode)
+			if !shouldFail {
```

Comment:
We should do the same Eventuallies but with fail instead of success, meaning that we should reach a final state for vmim.

Reply:
that's exactly what I'm doing. Stay posted.

Reply:
To be precise, we can't use the same `Eventually` blocks but switch their state because virt-controller never fills the `vmi.Status.MigrationState` if the migration fails.

Best I can do is check the VMIM object phase is failed.